packages/vm/src/index.ts

```diff
@@ -424,7 +429,18 @@ export default class VM extends AsyncEventEmitter {
    * available for VM execution
    */
   getActiveOpcodes(): OpcodeList {
-    return getOpcodesForHF(this._common, this._customOpcodes)
+    return getOpcodesForHF(this._common, this._customOpcodes).opcodes
+  }
+
+  /**
+   * Update the opcode data
+   * TODO: make this EIP-friendly
+   */
+  updateActiveOpcodes() {
+    const data = getOpcodesForHF(this._common, this._customOpcodes)
+    this._opcodes = data.opcodes
+    this._dynamicGasHandlers = data.dynamicGasHandlers
+    this._handlers = data.handlers
```

Comment:
This method shouldn't be exposed to the users I guess but rather be kept internally (and therefore made `private` and renamed to `_updateActiveOpcodes()` I guess (due to our current naming scheme within the VM)).

The opcodes + handlers just should be up-to-date when changes occur, and I guess this is also happening in the current state of the code, e.g. by being called on a HF-changed event.

While eventually being a bit less clean I guess I would even be a fan to "internalize" this functionality into `getActiveOpcodes`, so basically doing all of this functionality inside that function. Practically it comes out to the same result and then we do not have another `_updateActiveOpcodes()` function exposed to JS users (where this is not hidden but just visible e.g. on auto-completion in the Node CLI).

Reply:
Ah, yes, `getOpcodesForHF` likely makes even more sense for this.

(and then also maybe leave the existing return type for this function (this should then be possible, right?)? I am a bit afraid that someone might use this function directly for whatever purposes)

Reply:
This should be addressed now!